### ISSUE TITLE
Pylance stating module not found while module is usable in later code.

### ISSUE THREAD

**Issue by Syndicateeee** [microsoft/pylance-release#4976]
## Environment data

-   Language Server version: v2023.10.20
-   OS and version: Ubuntu 20.04
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda: Python 3.11.6

```python
from __future__ import absolute_import,division,print_function,unicode_literals

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from IPython.display import clear_output
from six.moves import  urllib

import tensorflow.compat.v2.feature_column as fc

import tensorflow as tf
```

## Expected behavior

Pylance should be detecting the imports as being correct because i can use them in my code with no problem.

## Actual behavior

Pylance is not detecting the imports of submodules `six.moves` and `tensorflow.compat.v2.feature_column` as being correct.

Note: I first discovered this error in my Jupyter notebook which also had the kernel set to my anaconda environment but i then tried it in a normal .py file and the same thing happened. Also, when I import tensorflow alone and then do tensorflow.compat.v2.... no error.

## Logs

```
Import "tensorflow.compat.v2.feature_column" could not be resolvedPylance[reportMissingImports](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportMissingImports)
```

**Comment by debonte** [comment]
What packages (and versions) do you have installed in your environment?

**Comment by Syndicateeee** [comment]
<details>
<summary>here is my pip list</summary>
absl-py                       2.0.0
anyio                         4.0.0
appdirs                       1.4.4
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
arrow                         1.3.0
asttokens                     2.4.0
astunparse                    1.6.3
async-lru                     2.0.4
attrs                         23.1.0
Babel                         2.13.0
backcall                      0.2.0
backports.functools-lru-cache 1.6.5
beautifulsoup4                4.12.2
bleach                        6.1.0
Brotli                        1.1.0
cached-property               1.5.2
cachetools                    5.3.1
certifi                       2023.7.22
cffi                          1.16.0
charset-normalizer            3.3.0
click                         8.1.7
comm                          0.1.4
contourpy                     1.1.1
cycler                        0.12.1
debugpy                       1.8.0
decorator                     5.1.1
defusedxml                    0.7.1
entrypoints                   0.4
exceptiongroup                1.1.3
executing                     1.2.0
fastjsonschema                2.18.1
flatbuffers                   23.5.26
fonttools                     4.43.1
fqdn                          1.5.1
gast                          0.4.0
google-auth                   2.23.3
google-auth-oauthlib          1.0.0
google-pasta                  0.2.0
googleapis-common-protos      1.56.4
grpcio                        1.59.0
grpcio-status                 1.59.0
h5py                          3.10.0
idna                          3.4
importlib-metadata            6.8.0
importlib-resources           6.1.0
ipykernel                     6.25.2
ipython                       8.16.1
isoduration                   20.11.0
jedi                          0.19.1
Jinja2                        3.1.2
json5                         0.9.14
jsonpointer                   2.4
jsonschema                    4.19.1
jsonschema-specifications     2023.7.1
jupyter_client                8.4.0
jupyter_core                  5.4.0
jupyter-events                0.8.0
jupyter-lsp                   2.2.0
jupyter_server                2.8.0
jupyter_server_terminals      0.4.4
jupyterlab                    4.0.7
jupyterlab-pygments           0.2.2
jupyterlab_server             2.25.0
keras                         2.13.1
kiwisolver                    1.4.5
libclang                      16.0.6
Mako                          1.2.4
Markdown                      3.5
MarkupSafe                    2.1.3
matplotlib                    3.8.0
matplotlib-inline             0.1.6
mistune                       3.0.1
nbclient                      0.8.0
nbconvert                     7.9.2
nbformat                      5.9.2
nest-asyncio                  1.5.8
notebook                      7.0.6
notebook_shim                 0.2.3
numpy                         1.24.3
nvidia-cublas-cu12            12.2.5.6
nvidia-cuda-nvrtc-cu12        12.2.140
nvidia-cuda-runtime-cu12      12.2.140
nvidia-cudnn-cu12             8.9.4.25
oauthlib                      3.2.2
opt-einsum                    3.3.0
overrides                     7.4.0
packaging                     23.2
pandas                        2.1.1
pandocfilters                 1.5.0
parso                         0.8.3
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.3.0
pip                           23.2.1
pip-autoremove                0.10.0
pkgutil_resolve_name          1.3.10
platformdirs                  3.11.0
prometheus-client             0.17.1
prompt-toolkit                3.0.39
protobuf                      4.24.4
psutil                        5.9.5
psycopg2                      2.9.3
ptyprocess                    0.7.0
pure-eval                     0.2.2
pyasn1                        0.5.0
pyasn1-modules                0.3.0
pycparser                     2.21
pycuda                        2022.2.2
Pygments                      2.16.1
pyparsing                     3.1.1
pyqt5-plugins                 5.15.9.2.3
PyQt5-Qt5                     5.15.2
PyQt5-sip                     12.13.0
PySocks                       1.7.1
python-dateutil               2.8.2
python-dotenv                 1.0.0
python-json-logger            2.0.7
pytools                       2023.1.1
pytz                          2023.3.post1
PyYAML                        6.0.1
pyzmq                         25.1.1
qt5-applications              5.15.2.2.3
qt5-tools                     5.15.2.1.3
referencing                   0.30.2
requests                      2.31.0
requests-oauthlib             1.3.1
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rpds-py                       0.10.6
rsa                           4.9
Send2Trash                    1.8.2
setuptools                    68.2.2
six                           1.16.0
sklearn                       0.0.post10
sniffio                       1.3.0
soupsieve                     2.5
stack-data                    0.6.2
tensorboard                   2.13.0
tensorboard-data-server       0.7.1
tensorflow                    2.13.1
tensorflow-estimator          2.13.0
tensorflow-io-gcs-filesystem  0.34.0
tensorrt-bindings             8.6.1
tensorrt-libs                 8.6.1
termcolor                     2.3.0
terminado                     0.17.1
tinycss2                      1.2.1
tomli                         2.0.1
tornado                       6.3.3
traitlets                     5.11.2
types-python-dateutil         2.8.19.14
typing_extensions             4.5.0
typing-utils                  0.1.0
tzdata                        2023.3
uri-template                  1.3.0
urllib3                       2.0.6
wcwidth                       0.2.8
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.6.4
Werkzeug                      3.0.0
wheel                         0.40.0
wrapt                         1.15.0
zipp                          3.17.0
</details>

**Comment by debonte** [comment]
Thanks for the additional info. I was able to reproduce this. Note that they are two different diagnostics.

- For `six.moves` the diagnostic is "reportMissingModuleSource" indicating that we found stubs for the module (in typeshed in this case), but could not find the implementation. This issue is a duplicate of https://github.com/microsoft/pylance-release/issues/1343.
- For `tensorflow.compat...` the diagnostic is "reportMissingImports" indicating that we found neither a stub nor an implementation. This issue is covered in https://github.com/microsoft/pyright/issues/2078 (see "problem 1"). Eric states that there is no `tensorflow.compat` submodule, which is true from a static analysis standpoint, but I my impression from digging around in their code is that it is created dynamically at runtime. There's a suggested workaround [here](https://github.com/microsoft/pyright/issues/2078#issuecomment-878817510). If that doesn't work, I'd suggest filing an issue on `tensorflow` requesting better support for static analysis tools.

Pylance is behaving as designed in both cases.

**Comment by Syndicateeee** [comment]
thanks ill have a look at this!

**Comment by MehdiaHumais** [comment]
my apis are not working, i cheak them all, they all are free

**Comment by rekhakolaji** [comment]
<img width="930" height="222" alt="Image" src="https://github.com/user-attachments/assets/155b5e94-e8be-4f40-8efc-c6123c1e1058" /> its showing like this will u please  give me suggestion to fix it

**Comment by rchiodo** [comment]
That should just be a problem in Pylance. tensorflow and keras dynamically import things and mess up Pylance's analysis. 

See this issue for more information:
https://github.com/keras-team/keras/issues/19779

**Comment by medhimanjyoti1-svg** [comment]
![Image](https://github.com/user-attachments/assets/ef283cbc-e573-4496-a008-6bdb1ec0b1c7)

![Image](https://github.com/user-attachments/assets/6690bde4-08c8-448d-90a4-e3c299c4d322)
kindly guide why this error is showing

**Comment by rchiodo** [comment]
@medhimanjyoti1-svg please see this for more information:
https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportMissingImports.md
